sale_coupon_multi_use/models/sale_order.py

```diff
@@ -0,0 +1,14 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    """Extend to modify action_confirm for multi-use coupons."""
+
+    _inherit = "sale.order"
+
+    def action_confirm(self):
```

Comment:
do we need to pass order in action_cancel and drop all consumption lines?

Reply:
Hm, looks like yes. Need to drop lines, because coupon is cleared